internal/wasm/store.go

```diff
@@ -183,16 +183,6 @@ func (m *ModuleInstance) buildExports(exports map[string]*Export) {
 		switch exp.Type {
 		case ExternTypeFunc:
 			ei = &ExportInstance{Type: exp.Type, Function: m.Functions[index]}
-			// The module instance of the host function is a fake that only includes the function and its types.
-			// We need to assign the ModuleInstance when re-exporting so that any memory defined in the target is
-			// available to the wasm.ModuleContext Memory.
-			//
-			// TODO(adrian): shouldn't this be race if multiple modules re-exporting?
```

Comment:
Note: this also stops mutating an imported function when exporting it. Notice all the other tests are still passing.